src/components/ArticleList.jsx

```diff
@@ -8,26 +8,70 @@ function ArticleList() {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const { topic } = useParams();
+  const [sort_by, setSort_by] = useState("created_at");
+  const [order, setOrder] = useState("DESC");
 
   useEffect(() => {
     setIsError(false);
-    setIsLoading(true);
-    getAllArticles(topic)
-      .then((response) => {
-        setArticles(response.data.articles);
-        setIsLoading(false);
-      })
-      .catch((error) => {
-        setIsError(true);
-        setIsLoading(false);
+    if (sort_by === "comment_count") {
+      setArticles(sortByComments(order, articles));
+    } else {
+      setIsLoading(true);
+      getAllArticles(topic, sort_by, order)
+        .then((response) => {
+          setArticles(response.data.articles);
+          setIsLoading(false);
+        })
+        .catch((error) => {
+          setIsError(true);
+          setIsLoading(false);
+        });
+    }
+  }, [topic, sort_by, order]);
+
+  function sortByComments(sortOrder, articlesToSort) {
+    const sortedArticles = [...articlesToSort];
+    if (sortOrder === "DESC") {
+      sortedArticles.sort((a, b) => {
+        return b.comment_count - a.comment_count;
+      });
+    } else {
+      sortedArticles.sort((a, b) => {
+        return a.comment_count - b.comment_count;
       });
-  }, []);
+    }
+    return sortedArticles;
+  }
```

Comment:
As you are passing in the articles and order as arguments this function could be extracted out as a utility function